src/System.IO.Pipelines/src/System/IO/Pipelines/Pipe.cs

```diff
@@ -488,11 +511,6 @@ private void AdvanceReader(BufferSegment consumedSegment, int consumedIndex, Buf
                 // but only if writer is not completed yet
                 if (examinedEverything && !_writerCompletion.IsCompleted)
                 {
-                    // Prevent deadlock where reader awaits new data and writer await backpressure
-                    if (!_writerAwaitable.IsCompleted)
```

Comment:
Can we Debug.Assert that _writerAwaitable.IsCompleted?

Reply:
Sure